persist-cli/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -134,17 +144,74 @@ public void execute() {
+                HashMap<String, String> persistConfig = readPersistConfigurations(
+                        Paths.get(this.sourcePath, "Ballerina.toml"));
+
+                if (!persistConfig.get("module").equals(packageName)) {
+                    String submodule = persistConfig.get("module").split("\\.")[1];
+                    if (!ProjectUtils.validateModuleName(submodule)) {
+                        errStream.println("ERROR: invalid module name : '" + submodule + "' :\n" +
+                                "module name can only contain alphanumerics, underscores and periods");
+                        return;
+                    } else if (!ProjectUtils.validateNameLength(submodule)) {
+                        errStream.println("ERROR: invalid module name : '" + submodule + "' :\n" +
+                                "maximum length of module name is 256 characters");
+                        return;
+                    }
+                    generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY,
+                            submodule);
+                }
+                if (!Files.exists(generatedSourceDirPath)) {
+                    try {
+                        if (!persistConfig.get("module").equals(packageName) && !Files.exists(
+                                Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY))) {
+                            Files.createDirectory(Paths.get(this.sourcePath,
+                                    BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY).toAbsolutePath());
+
+                        }
+                        Files.createDirectory(generatedSourceDirPath.toAbsolutePath());
+                    } catch (IOException e) {
+                        errStream.println("ERROR: failed to create the generated directory. " + e.getMessage());
+                        return;
+                    }
+                }
+
+                Path databaseConfigPath = generatedSourceDirPath.resolve(PATH_CONFIGURATION_BAL_FILE);
+                if (!Files.exists(databaseConfigPath)) {
+                    try {
+                        generateConfigurationBalFile(generatedSourceDirPath);
+                        errStream.printf(
+                                "Created database_configurations.bal file inside `%s` module in generated directory.%n",
+                                fileName.equals(packageName) ? "default" : fileName);
+                    } catch (BalException e) {
+                        errStream.println("ERROR: failed to generate the database_configurations.bal file. "
+                                + e.getMessage());
+                        return;
+                    }
+                }
+
+                if (!Files.exists(Paths.get(this.sourcePath, CONFIG_SCRIPT_FILE).toAbsolutePath())) {
+                    createConfigTomlFile(schemaFilePaths, packageName);
```

Comment:
why do we need to pass schemaFilePaths to update Config.toml file?

Reply:
updated in [566e36d](https://github.com/ballerina-platform/persist-tools/pull/148/commits/566e36dd84d453026fbf2506266f34668c08e1af)